dashboard/app/models/channel_token.rb

```diff
@@ -20,15 +20,24 @@ class ChannelToken < ActiveRecord::Base
   belongs_to :user
   belongs_to :level
 
+  # @param [Level] level The level associated with the channel token request.
+  # @param [User] user The user associated with the channel token request.
+  # @param [String] ip The IP address making the channel token request.
+  # @param [StorageApps] storage_app The storage app associated with the channel token request.
+  # @param [Hash] data
+  # @return [ChannelToken] The channel token (new or existing).
   def self.find_or_create_channel_token(level, user, ip, storage_app, data = {})
     # If `create` fails because it was beat by a competing request, a second
     # `find_by` should succeed.
     Retryable.retryable on: [Mysql2::Error, ActiveRecord::RecordNotUnique], matching: /Duplicate entry/ do
       # your own channel
-      find_or_create_by!(level: level.host_level, user: user) do |ct|
+      channel_token = find_or_initialize_by(level: level.host_level, user: user) do |ct|
         # Get a new channel_id.
         ct.channel = create_channel ip, storage_app, data: data
+        ct.storage_app_id = (storage_decrypt_channel_id ct.channel).second
       end
+      channel_token.save!
+      channel_token
```

Comment:
For my own education, what is the reason for changing this from `find_or_create_by!` to `find_or_initialize_by` ?